ivadomed/models.py

```diff
@@ -573,18 +573,18 @@ def __init__(self, n_features, n_channels, n_hid=64):
         self.linear3 = nn.Linear(n_hid // 4, n_channels * 2)
 
     def forward(self, x, shared_weights=None):
-        x = self.linear1(x)
-        x = self.sig(x)
-
         if shared_weights is not None:  # weight sharing
-            self.linear2.weight = shared_weights
+            self.linear1.weight = shared_weights[0]
+            self.linear2.weight = shared_weights[1]
 
+        x = self.linear1(x)
+        x = self.sig(x)
```

Comment:
I can't remember why, two years ago, I put these lines before the weight sharing. What are your thoughts about this?

Reply:
`self.linear1` used to not have shared weights, so putting before or after the weight sharing would give the same results. With the new code, we have to extract the shared weights, then use the layer. If we put `x = self.linear1(x)` before the weight sharing, the behavior would be the same as before (no shared weights for the first layer).

Reply:
Makes full sense!